### ISSUE TITLE
Deprecated SASS syntax for upcoming Dart SASS

### ISSUE THREAD

**Issue by mistic100** [ag-grid/ag-grid#3623]
Follow up of #3597

**I'm submitting a ...**  (check one with "x")
```
[X] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[] support request => see 'Requesting Community Support'
```

Versions : 
```json
    "ag-grid-angular": "22.1.1",
    "ag-grid-community": "22.1.1",
    "ag-grid-enterprise": "22.1.1",
    "@angular/common": "8.2.14",
    "@angular/core": "8.2.14",
    "@angular/compiler": "8.2.14",
```

Rhe following warnings are displayed when building the app :
```
DEPRECATION WARNING: As of Dart Sass 2.0.0, !global assignments won't be able to
declare new variables. Consider adding `$ag-secondary-font-weight: null` at the root of the
stylesheet.

    ╷
211 │     $ag-secondary-font-weight: $secondary-font-weight !global;
    │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ╵
    node_modules/@ag-grid-community/all-modules/dist/styles/ag-theme-base/vars/_ag-theme-base-import-legacy-vars.scss 211:5        @import
    node_modules/@ag-grid-community/all-modules/dist/styles/ag-theme-material/vars/_ag-theme-material-import-legacy-vars.scss 3:9  @import
    node_modules/@ag-grid-community/all-modules/dist/styles/ag-theme-material/vars/_ag-theme-material-vars.scss 1:9                @import
    node_modules/@ag-grid-community/all-modules/dist/styles/ag-theme-material/sass/ag-theme-material.scss 1:9                      @import
```

The problems are in the theme files where all global variable declarations must be declared before-hand
https://github.com/ag-grid/ag-grid/blob/master/dist/styles/ag-theme-base/vars/_ag-theme-base-import-legacy-vars.scss

Relevant changelog https://github.com/sass/dart-sass/blob/master/CHANGELOG.md#1172

**there is not need a for a reproducible scenario, the syntax problem is clear and identified**

Thank you

**Comment by AhmedAGadir** [comment]
Hi,

Your ticket has been flagged as in_zendesk, it means that we recognize this as an issue worth investigating so we have moved it to our official support channel for customers (zendesk)

We will carry on there with the investigation and we will update you as soon as possible.

Thanks

**Comment by dpraul** [comment]
Looks like this is a result of [the legacy theme variable import](https://github.com/ag-grid/ag-grid/blob/996ffcdcb828ffa3448d57fa919feb9eb465df15/community-modules/core/src/styles/ag-theme-base/vars/_ag-theme-base-import-legacy-vars.scss#L4). If you're on the newest version of `ag-grid` and have migrated your themes to use the [new theme variables](https://github.com/ag-grid/ag-grid/blob/996ffcdcb828ffa3448d57fa919feb9eb465df15/community-modules/core/src/styles/ag-theme-base/vars/_ag-theme-base-vars.scss) (e.g. `background-color` becomes `ag-background-color`), then you should be able to add the following line to your theme variables and the problem lines should never be hit:

```scss
$ag-suppress-legacy-var-import: true;
```

**Comment by mistic100** [comment]
Thank's for the tip

**Comment by vnovikov-clgx** [comment]
Can some one tell me how to fix it?

**Comment by davis4acca** [comment]
Hi,

Thank you for your email. We are aware of this issue and have a feature request in our backlog.

This is the reference and summary of the relevant issue:

AG-3797 - Investigate improving DART support for Sass build

You can follow the status of feature requests, bugs and releases using our pipeline:
https://www.ag-grid.com/ag-grid-pipeline/

If you are an ag-Grid Enterprise Customer, please raise this in our Zendesk support system - contact info@ag-grid.com for access. This is our primary channel for Support.

**Comment by lk77** [comment]
Hello,

any news on this ?

**Comment by mistic100** [comment]
bad bot

Sent from MailDroid

-----Original Message-----

**Comment by mistic100** [comment]
I hate this bot

**Comment by hwebb** [comment]
Just a comment to warn you that I was about to purchase the enterprise version as I was quite impressed with the functionalities. But this issue and the huge delay for the fix is giving me cold feet, and I'll start checking for the competition now. There may be many others like me.

**Comment by makinggoodsoftware** [comment]
Hi guys,

Just a note to let you know that we are looking into this now.

You should hear on this thread soon!

Thanks for your patience

**Comment by makinggoodsoftware** [comment]
Hi,

Thank you for your email. We are aware of this issue and have a feature request in our backlog:

You can follow the status of feature requests, bugs and releases using our pipeline:
https://www.ag-grid.com/ag-grid-pipeline/

If you are an ag-Grid Enterprise Customer, please raise this in our Zendesk support system - contact info@ag-grid.com for access. This is our primary channel for Support.



This is the reference and summary of the relevant issue:

AG-3797 - Investigate improving DART support for Sass build

**Comment by julmot** [comment]
@makinggoodsoftware Where can I see the status on this exactly?

![AG-3797](https://user-images.githubusercontent.com/8310677/124468256-2c6ce180-dd99-11eb-8eb0-9ef0ed84f52f.PNG)

**Comment by gportela85** [comment]
@julmot If a ticket is not listed under the pipeline it probably has been fixed, so you should check the changelog. https://www.ag-grid.com/ag-grid-changelog/

**Comment by julmot** [comment]
@gportela85 Thanks for the hint.

According to that, the fix was introduced in v23.1.1. I'm on version v25.3.0, so I would assume that no warnings appear. But this isn't the case. Is my assumption correct, that this is only the investigation that was part of this ticket, not the actual fix? If so, where can I see the status on the actual fix?

**Comment by gportela85** [comment]
@julmot The warnings for Sass Dart keep changing as the versions change. Back when this ticket was created, the warnings were about the `!global` flag.
Recently, new warnings have appeared about the `division` symbol being deprecated, and it was tracked here: https://github.com/ag-grid/ag-grid/issues/4543

If you see different warnings, file a new issue report. However, I have changed our build process to use Dart Sass instead of Node Sass, so I think we will be on top of it from now on.
